caas/kubernetes/provider/k8s.go

```diff
@@ -1192,6 +1190,22 @@ func (k *kubernetesClient) EnsureService(
 	return nil
 }
 
+func validateDeploymentType(t caas.DeploymentType, params *caas.ServiceParams, workloadSpec *workloadSpec) error {
+	if t != caas.DeploymentStateful {
+		if workloadSpec.Service != nil && workloadSpec.Service.ScalePolicy != "" {
+			return errors.NewNotValid(nil, fmt.Sprintf("ScalePolicy is only supported for %s applications", caas.DeploymentStateful))
+		}
+	}
+
+	if t != caas.DeploymentStateful && len(params.Filesystems) > 0 {
+		return errors.NewNotValid(nil, fmt.Sprintf(
+			"metadata.storage is only supported for deployment type %q, please define storage using volumeConfig in containers",
```

Comment:
charm defined storage is only supported for deployment type %q, please instead define storage using volumeConfig in containers